index.ts

```diff
@@ -0,0 +1,186 @@
+    log('accessibility:', colorScore(processScore(accessibility.score)));
+    log('bestPractices:', colorScore(processScore(bestPractices.score)));
+    log('seo:', colorScore(processScore(seo.score)), '\n');
+};
+
+interface CustomResults {
+    [website: string]: {
+        [platform: string]: Array<{
+            performance: number;
+            accessibility: number;
+            'best-practices': number;
+            seo: number;
+        }>;
+    };
+}
+
+export const runPsi = async (options: Options) => {
+    console.log('Running PageSpeed Insights...');
+    const numRuns = options.num ?? NUM_RUNS;
+
+    options.mobile && platforms.filter((url) => url === 'mobile');
+
+    let customResults: CustomResults = {};
+
+    for (const url of options.websites) {
+        for (const platform of platforms) {
+            const results = [];
+            for (let i = 0; i < numRuns; i++) {
+                // To prevent Google PSI API from returning the previous cached result
+                const urlWithRun = `${url}?run=${uuid()}`;
+                const key = options.key ?? process.env.API_KEY ?? '';
+                console.log(
+                    'Running',
+                    colorPlatform(platform),
+                    `Lighthouse audit #${i + 1} of ${numRuns}`,
+                    `${options.local ? 'locally' : 'on Google'}. \n${urlWithRun}:`
+                );
+
+                try {
+                    const runnerResult = options.local ? localRun(urlWithRun) : await psiRun(urlWithRun, key, platform);
+                    results.push(runnerResult);
+                    singleOutput(runnerResult);
+                } catch (e) {
+                    console.error(e);
+                    continue;
+                }
+            }
+
+            const median = computeMedianRun(results);
+            console.log(
+                `Median performance score on`,
+                colorPlatform(platform),
+                'was',
+                colorScore(processScore(median.categories.performance.score), true),
+                '\n'
+            );
+
+            customResults = results.reduce((acc, curr) => {
+                const { performance, seo, accessibility, 'best-practices': bestPractices } = curr.categories;
+                acc[url] = {
+                    ...customResults[url],
+                    [platform]: [
+                        ...(acc[url]?.[platform] ?? []),
+                        {
+                            performance: processScore(performance.score),
+                            accessibility: processScore(accessibility.score),
+                            'best-practices': processScore(bestPractices.score),
+                            seo: processScore(seo.score),
+                        },
+                    ],
+                };
+                return acc;
+            }, customResults);
+        }
+    }
+    fs.mkdirSync('results', { recursive: true });
```

Comment:
What does `recursive: true` do here?

Reply:
I was having issues where it wasn't actually making the directory if it didn't exists, and adding this option in fixed it after reading through [this SO post](https://stackoverflow.com/questions/31645738/how-to-create-full-path-with-nodes-fs-mkdirsync). It is a bit confusing, I agree